### PR TITLE
OPS-2973: Avoid ifup/down when adding routes and addded new options.

### PR DESCRIPTION
**Pull request by Bezzi**
Added option to restart all NIC or not when configuring an interface.
$restart_all_nic = true, (Meaning that will restart all interfaces when configuring a given interface).

Added custom scripts to avoid ifdown/up when adding a IPv4/v6 route in hierafile, and modified existing z90ifup/ifdown generated scripts to support IPv6.

### CODE REVIEW

**Comment by Bezzi** [comment]
Maybe the file "templates/route_down-Debian.erb" could be replaced by using the command "ip route / ip -6 route flush dev br0" erasing every root from the routing table, even those not included in the hierafile.

**Comment by pedrorochagoncalves** [comment]
@Bezzi looks good to me! My only question is, if you add an IPv6 address to an interface, it's going to restart the interface right? Can we make it so that it doesn't restart the interface? We should be able to configure IPs without restarting interfaces.

**Comment by Bezzi** [comment]
@pedrorochagoncalves This way the interface is only restarted when is first being configured, and I added a boolean that allows to restart only the interface being configured and not all of them (It's deactivated by default just to keep the same functionality as before). And when a route is added either IPv4/v6 because i disabled the notify on ifdown/ifup in routes.pp the interface won't be restarted, and the exec { 'refresh-routes-${name}': will be executed adding the routes without doing ifdown/ifup !

File: templates/route_down-Debian.erb [hunk 1]
```diff
@@ -3,7 +3,5 @@
 ### File managed by Puppet
 #
 if [ "$IFACE" = "<%= @interface -%>" ]; then
-<%- (0..(@ipaddress.length-1)).each do |id| -%>
-    ip route del <%= @ipaddress[id] %>/<%= @netmask[id] %> via <%= @gateway[id] %> dev <%= @interface %>
-<%- end -%>
-fi
+    sudo ip -6 route flush dev <%= @interface -%>
```
Comment (pedrorochagoncalves) [inline templates/route_down-Debian.erb]: Afaik these scripts already run as root. Don't think sudo is needed here.

Comment (Bezzi) [reply templates/route_down-Debian.erb]: Oh right. I added them because I wasn't sure. I'll remove it

**Pull request merged**